app/controllers/BillsController.scala

```diff
@@ -41,15 +44,22 @@ class BillsController @Inject()(implicit val config: FrontendAppConfig,
                                ) extends BaseController {
 
   val viewCrystallisedCalculations: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino andThen retrieveIncomeSources).async {
-    implicit user =>
-      implicit val sources: IncomeSourcesModel = user.incomeSources
-
-      for {
-        lastTaxCalcs <- calculationService.getAllLatestCalculations(user.nino, sources.orderedTaxYears)
-      } yield {
-          Logger.debug(s"[BillsController][viewCrystallisedCalculations] Retrieved Last Tax Calcs With Year response: $lastTaxCalcs")
-          if (lastTaxCalcs.exists(_.isErrored)) itvcErrorHandler.showInternalServerError
-          else Ok(views.html.bills(lastTaxCalcs.filter(_.matchesStatus(Crystallised))))
-      }
+    implicit user => if(config.features.billsEnabled()) renderView else redirectToHome
+  }
+
+  private[BillsController] def redirectToHome: Future[Result] = {
```

Comment:
We could refactor this to a common method that is on the `BaseController` trait. As there will be a number of controllers which need to redirect to home. There's an example on one of my PRs.